addons/account/models/account_move.py

```diff
@@ -3305,6 +3312,17 @@ def _check_total_amount(self, amount_total):
                 totals['amount_total'] = amount_total
                 self.tax_totals = totals
 
+    def _round_total_amount(self, amount_total):
+        """
+        Rounds total amount of move using _check_total_amount. Verifies that the total
+        amount indeed needs rounding, so larger differences between the desired amount
+        total and the current amount total are not wrongly treated as a rounding error.
+        The currency_id.rounding is used as a threshold to determine that.
+        """
+        difference = abs(amount_total - self.amount_total)
+        if float_compare(difference, self.currency_id.rounding, precision_rounding=self.currency_id.rounding) == 0:
```

Comment:
```suggestion
        if not self.currency_id.is_zero(difference):
```

Reply:
(note that the suggestion is doing the opposite as the original diff, not sure if wanted)